connection.go

```diff
@@ -48,6 +49,26 @@ type mysqlConn struct {
 	finished chan<- struct{}
 	canceled atomicError // set non-nil if conn is canceled
 	closed   atomicBool  // set when conn is closed, before closech is closed
+
+	// for killing query after timeout
+	id int
+	d MySQLDriver
+	dsn string
+}
+
+func (mc *mysqlConn) killSelfQuery() error {
+	killConn, killErr := mc.d.Open(mc.dsn)
+	if killErr != nil {
+		return killErr
+	}
+	defer killConn.Close()
+	return killConn.(*mysqlConn).killQuery(mc.id)
+}
+
+func (mc *mysqlConn) killQuery(connId int) error {
```

Comment:
What is purpose of this function? this code can be also implemented in previous one.

Reply:
removed